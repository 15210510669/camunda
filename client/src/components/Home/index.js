/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
