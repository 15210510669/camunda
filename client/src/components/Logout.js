/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    mightFail(
      get('api/authentication/logout'),
      () => {
        addNotification({text: t('navigation.logoutSuccess')});
        history.replace('/');
      },
      () => {
        addNotification({text: t('navigation.logoutFailed'), type: 'error'});
        history.replace('/');
      }
    );
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
