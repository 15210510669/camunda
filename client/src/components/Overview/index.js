export {default as Overview} from './Overview';
export {default as Alerts} from './Alerts';
