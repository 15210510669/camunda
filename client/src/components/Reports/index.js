export {default as Reports} from './Reports';
export {default as Report} from './Report';
export {Filter} from './filter'
