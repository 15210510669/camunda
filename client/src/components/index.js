export {Footer} from './Footer';
export {Header, LogoutButton} from './Header';
export {Home} from './Home';
export {Login} from './Login';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboard';
export {Dashboards} from './Dashboards';
export {Report} from './Report';
export {Reports} from './Reports';
