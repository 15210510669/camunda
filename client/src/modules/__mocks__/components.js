export {default as Button} from '../components/__mocks__/Button';
export {default as Input} from '../components/__mocks__/Input';
export {default as Modal} from '../components/__mocks__/Modal';
export {default as Popover} from '../components/__mocks__/Popover';
export {default as ControlGroup} from '../components/__mocks__/ControlGroup';
export {default as ShareEntity} from '../components/__mocks__/ShareEntity';
export {default as DashboardView} from '../components/__mocks__/DashboardView';
export {default as Icon} from '../components/__mocks__/Icon';
export {default as Dropdown} from '../components/__mocks__/Dropdown';
export {default as Message} from '../components/__mocks__/Message';
export {default as ReportView} from '../components/__mocks__/ReportView';
export {default as Select} from '../components/__mocks__/Select';
export {default as BPMNDiagram} from '../components/__mocks__/BPMNDiagram';
export {default as TargetValueBadge} from '../components/__mocks__/TargetValueBadge';
export {default as Table} from '../components/__mocks__/Table';
export {default as Logo} from '../components/__mocks__/Logo';
export {default as ErrorMessage} from '../components/__mocks__/ErrorMessage';
