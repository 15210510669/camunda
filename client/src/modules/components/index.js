/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {Table} from './Table';
export {Button} from './Button';
export {Modal} from './Modal';
export {Message} from './Message';
export {Dropdown} from './Dropdown';
export {Input} from './Input';
export {Select} from './Select';
export {ButtonGroup} from './ButtonGroup';
export {BPMNDiagram, ClickBehavior} from './BPMNDiagram';
export {ControlGroup} from './ControlGroup';
export {CopyToClipboard} from './CopyToClipboard';
export {ReportRenderer} from './ReportRenderer';
export {ErrorBoundary} from './ErrorBoundary';
export {Logo} from './Logo';
export {Icon} from './Icon';
export {ActionItem} from './ActionItem';
export {ShareEntity} from './ShareEntity';
export {Switch} from './Switch';
export {Popover} from './Popover';
export {DashboardView} from './DashboardView';
export {DashboardObject} from './DashboardView';
export {DefinitionSelection} from './DefinitionSelection';
export {TargetValueBadge} from './TargetValueBadge';
export {ErrorPage} from './ErrorPage';
export {ErrorMessage} from './ErrorMessage';
export {Typeahead} from './Typeahead';
export {Labeled} from './Labeled';
export {LabeledInput} from './LabeledInput';
export {TypeaheadMultipleSelection} from './TypeaheadMultipleSelection';
export {DatePicker} from './DatePicker';
export {LoadingIndicator} from './LoadingIndicator';
export {ConfirmationModal} from './ConfirmationModal';
export {ColorPicker} from './ColorPicker';
export {EntityNameForm} from './EntityNameForm';
export {CollectionsDropdown} from './CollectionsDropdown';
export {EditCollectionModal} from './EditCollectionModal';
