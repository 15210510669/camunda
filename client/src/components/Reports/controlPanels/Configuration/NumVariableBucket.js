/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React, {useCallback, useState} from 'react';

import {LabeledInput, Switch, Input, Message} from 'components';
import debounce from 'debounce';
import {numberParser} from 'services';
import {t} from 'translation';

export default function NumVariableBucket({
  report: {
    data: {
      configuration: {customNumberBucket},
      groupBy,
    },
  },
  onChange,
}) {
  const [sizeValid, setSizeValid] = useState(true);
  const [baseValid, setBaseValid] = useState(true);

  const applyChanges = useCallback(
    debounce((property, value, valid) => {
      if (valid) {
        onChange({customNumberBucket: {[property]: {$set: value}}}, true);
      }
    }, 800),
    []
  );

  if (
    groupBy?.type === 'variable' &&
    ['Integer', 'Double', 'Short', 'Long'].includes(groupBy.value?.type)
  ) {
    const {active, bucketSize, baseline} = customNumberBucket;

    return (
      <fieldset className="NumVariableBucket">
        <legend>
          <Switch
            checked={active}
            onChange={(evt) =>
              onChange({customNumberBucket: {active: {$set: evt.target.checked}}}, true)
            }
            label={t('report.config.bucketSize')}
          />
        </legend>
        <Input
          disabled={!active}
          isInvalid={!sizeValid}
          onChange={(evt) => {
            const valid = numberParser.isNonNegativeNumber(evt.target.value);
            setSizeValid(valid);
            applyChanges('bucketSize', evt.target.value, valid);
          }}
          defaultValue={bucketSize}
        />
        {!sizeValid && <Message error>{t('report.config.goal.invalidInput')}</Message>}
        <LabeledInput
          label={t('report.config.baseline')}
          disabled={!active}
          isInvalid={!baseValid}
          onChange={(evt) => {
            const valid = !isNaN(evt.target.value);
            setBaseValid(valid);
            applyChanges('baseline', evt.target.value, valid);
          }}
          defaultValue={baseline}
        />
        {!baseValid && <Message error>{t('report.config.invalidNumber')}</Message>}
      </fieldset>
    );
  }
  return null;
}
