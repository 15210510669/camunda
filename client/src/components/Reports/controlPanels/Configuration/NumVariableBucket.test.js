/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';
import {Input, LabeledInput} from 'components';

import NumVariableBucket from './NumVariableBucket';

const report = {
  data: {
    groupBy: {type: 'variable', value: {type: 'Integer'}},
    configuration: {
      customNumberBucket: {
        active: false,
        bucketSize: '10',
        baseline: '0',
      },
    },
  },
};

jest.mock('debounce', () => jest.fn((fn) => fn));

it('should render nothing if the current variable is not a number', () => {
  const node = shallow(
    <NumVariableBucket
      report={{data: {...report.data, groupBy: {type: 'variable', value: {type: 'Date'}}}}}
    />
  );

  expect(node).toMatchSnapshot();
});

it('should render bucket options', () => {
  const node = shallow(<NumVariableBucket report={report} />);

  expect(node).toMatchSnapshot();
});

it('should active the bucket size when enabling the switch', () => {
  const spy = jest.fn();

  const node = shallow(<NumVariableBucket report={report} onChange={spy} />);

  node.find('Switch').prop('onChange')({target: {checked: true}});

  expect(spy).toHaveBeenCalledWith({customNumberBucket: {active: {$set: true}}}, true);
});

it('should reevaluate the report when changing the size or baseline to a valid value', () => {
  const spy = jest.fn();

  const node = shallow(<NumVariableBucket report={report} onChange={spy} />);

  node.find(Input).prop('onChange')({target: {value: '-50'}});
  expect(spy).not.toHaveBeenCalled();
  expect(node.find('Message').dive()).toIncludeText('non-negative');

  node.find(LabeledInput).prop('onChange')({target: {value: '-1'}});

  expect(spy).toHaveBeenCalledWith({customNumberBucket: {baseline: {$set: '-1'}}}, true);
});
