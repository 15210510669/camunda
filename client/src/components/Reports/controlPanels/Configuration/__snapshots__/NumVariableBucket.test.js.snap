// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render bucket options 1`] = `
<fieldset
  className="NumVariableBucket"
>
  <legend>
    <Switch
      checked={false}
      label="Bucket Size"
      onChange={[Function]}
    />
  </legend>
  <ForwardRef(Input)
    defaultValue="10"
    disabled={true}
    isInvalid={false}
    onChange={[Function]}
  />
  <ForwardRef(LabeledInput)
    defaultValue="0"
    disabled={true}
    isInvalid={false}
    label="Baseline"
    onChange={[Function]}
  />
</fieldset>
`;

exports[`should render nothing if the current variable is not a number 1`] = `""`;
